### ISSUE TITLE
AssemblyScript Public Meeting #8 - March 25th, 2020

### ISSUE THREAD

**Issue by torch2424** [AssemblyScript/meta#26]
# Date and Time

This public meeting will take place: March 11th, 2020, 19:00 UTC (12:00 PM (Noon) US Pacific Standard Time UTC -8)

# General Agenda

* Agenda Items from this Github issue
* Additional in-meeting comments / discussion

Feel free to comment on this issue if you have any agenda items you would like to bring up. 
Meeting Notes will be placed on this issue for those who cannot make the meeting.

# Meeting Information

We will use Google Meet for our meetings. Our meeting room is: https://meet.google.com/ofw-zkpi-aek

Anyone who is contributing to the AssemblyScript project, building something with AssemblyScript, interested in the Assembly project or WebAssembly in general, is welcome to join! 😄

Notes will be taken by the host, and posted after the meeting. Notes are free to be edited through comments on the meeting notes at a later time.

**Comment by torch2424** [comment]
Let's talk about the time! Forgot to bring it up last time :joy:

**Comment by dcodeIO** [comment]
* Are we all fine with going all-in on removing `===` identity semantics as tracked by https://github.com/AssemblyScript/assemblyscript/pull/1111 once https://github.com/WebAssembly/binaryen/pull/2702 enables us to? [Twitter vote](https://mobile.twitter.com/AssemblyScript/status/1228638321305378818) was about even, and [one valid objection](https://github.com/AssemblyScript/assemblyscript/pull/1111#issuecomment-586629659) in the PR.

**Comment by jtenner** [comment]
We need a babel-like api to help support developers who want to do ast transforms. I know numerous devs who would benefit from such an api. Perhaps that could be another feature supported by aspect?

**Comment by jtenner** [comment]
I'm also blocked on master currently with wasi support waiting to release a 3.2 version of aspect because of the loader refactor, which has yet to release stable.

**Comment by jtenner** [comment]
Things I can work on: 

- loader getStaticArray methods
- loader getStaticArrayView methods
- loader allocStaticArray method

**Comment by torch2424** [comment]
Meeting Notes that we can all edit on during the meeting, will be moved here after: https://docs.google.com/document/d/1_1lv7xKo2fKFDADTAplP4R3p3Z4vloma1P9xx7Uqwpo/edit?usp=sharing

**Comment by torch2424** [comment]
# Notes

## New Meeting Time ( @torch2424  )

Agreed to move the meeting time one hour earlier. 

So that would be: 18:00 UTC (11:00 AM (Noon) US Pacific Standard Time UTC -8), 6pm @dcodeIO time :joy: 

## Triple Equals ( @dcodeIO  )

* Breaking change just for TS compatibility
* Has some interesting null checking implications, but it is quite an edge case that would be breaking.
* Tyepscript has a null type, but wasm doesn’t. So we will have to do some weird stuff.
* General thoughts: Let’s think more about general scaling dev experience vs. typescript compatibility. Most agnostic to this, but whatever dcode thinks would satisfy that.
* May be a good idea to just provide a null checking thing in the standard library, instead of depending on our triple equals
* @dcodeIO  will bring up a few examples and bring it up again :)

## Visitor API ( @jtenner @willemneal )

* Visitor API, will be super helpful, similar to babel
* Looking for some help getting this done
* https://github.com/willemneal/visitor-as
* Dcode mentions an AST helper may be blocked by missing Wasm / AS Closure support
* https://github.com/AssemblyScript/assemblyscript/tree/master/src/extra

### Near Protocol Examples ( @willemneal )

* Gitpod: https://examples.nearprotocol.com/
* TODO: Add to built with assemblyscript ( @torch2424 can add it )

## Loader functions for static array types ( @jtenner )

* Submit PR for as-loader static array types ( @jtenner )
* No objects from anyone :smile: 

## WasmByExample ( @torch2424 )

* Quality of life issues closed
* Classes in AS example by @battlelinegames

## Interface types ( @willemneal )

* Lets bring it up, maybe a seperate meeting

## Loader refactor ( @jenner )

* Would be nice to have a version slice soon with refactor stuff.
* Shopify folks would like the WASI stuff soon for as-pect.
* @dcodeIO - let’s do Wasi and triple equals all in 0.10.x .

* Community group ( @DuncanUszkay1 )

* Currently, between NEAR and Shopify
* Open to public, anyone welcome
* More Details to come later cc @DuncanUszkay1 and @willemneal ? 

## Language Server (Saul Cabrera)

* Interested in taking this up, scanned some issues, and saw the reasoning for piggy backing typescript.
* Will start the work, everyone super excited, wanting to help
* Will start on Saul's own repo, and can merge back in.

## Closure support ( @DuncanUszkay1 )

* Been busy, no progress yet
* But will try to make my work so far more public
* EDIT: Work was posted: https://gist.github.com/DuncanUszkay1/79f892101a3ae3291eb3c05a28bf4e3f

## Assertions ( @DuncanUszkay1 )

* @dcodeIO - we can definitely make this easier, we can chat  :smile:

**Comment by torch2424** [comment]
Feel free to continue the discussion or ask questions about this meeting here. 

Closing this issue, and opened an issue for the next meeting: #27

**Comment by torch2424** [comment]
@willemneal added the NEAR examples to the gitbook! :smile: :tada:
